stored_requests/backends/db_fetcher/fetcher_test.go

```diff
@@ -70,6 +93,22 @@ func TestPartialResponse(t *testing.T) {
 	assertHasData(t, storedReqs, "stored-req-id", "{}")
 }
 
+func TestPartialResponseFetchResponses(t *testing.T) {
+	mockQuery := "SELECT id, data, 'response' AS dataType FROM responses_table WHERE id IN (?, ?)"
+	mockReturn := sqlmock.NewRows([]string{"id", "data", "dataType"}).
+		AddRow("stored-resp-id", "{}", "response")
+
+	mock, fetcher := newFetcher(t, mockReturn, mockQuery, "stored-resp-id", "stored-resp-id-2")
+	defer fetcher.db.Close()
+
+	storedResponses, errs := fetcher.FetchResponses(context.Background(), []string{"stored-resp-id", "stored-resp-id-2"})
+
+	assertMockExpectations(t, mock)
+	assertErrorCount(t, 0, errs)
+	assertMapLength(t, 1, storedResponses)
+	assertHasData(t, storedResponses, "stored-resp-id", "{}")
+}
+
```

Comment:
Similar comment to above, where I think the TestGoodRepsonseFetchResponses, TestPartialResponse, and TestEmptyResponse tests could be consolidated into one table driven test.

Reply:
Done